op/model.py

```diff
@@ -423,6 +424,56 @@ def set_spec(self, spec, k8s_resources=None):
         self._backend.pod_spec_set(spec, k8s_resources)
 
 
+class StorageMapping(Mapping):
+    """Map of storage names to lists of Storage instances."""
+
+    def __init__(self, storage_names, backend):
+        self._backend = backend
+        self._storage_map = {storage_name: None for storage_name in storage_names}
+
+    def __contains__(self, key):
+        return key in self._storage_map
+
+    def __len__(self):
+        return len(self._storage_map)
+
+    def __iter__(self):
+        return iter(self._storage_map)
+
+    def __getitem__(self, storage_name):
+        storage_list = self._storage_map[storage_name]
+        if storage_list is None:
+            storage_list = self._storage_map[storage_name] = []
+            for storage_id in self._backend.storage_list(storage_name):
+                storage_list.append(Storage(storage_name, storage_id, self._backend))
+        return storage_list
+
+    def add(self, storage_name, count=1):
+        """Adds new storage instances of a given name.
+
+        Juju will notify the unit via <storage-name>-storage-attached events when it becomes available.
+        """
+        if storage_name not in self._storage_map:
+            raise ModelError(f'cannot add storage with {storage_name} as it is not present in the charm metadata')
+        self._backend.storage_add(storage_name, count)
+
+
+class Storage:
+    """A storage unit."""
```

Comment:
This comment can go I think. Doesn't add any meaningful info to the class name right above it.